infrastructure/index.ts

```diff
@@ -267,20 +261,65 @@ crawlDirectory(
     webContentsRootPath,
     (filePath: string) => {
         const relativeFilePath = filePath.replace(webContentsRootPath + "/", "");
+
+        // If the file is a .html file that contains a meta refresh redirect, configure the bucket object to be a
+        // webpage redirect to make S3 return a 301 response when the object is requested. This is done to allow
+        // #anchors in URLs to flow across to the destination page, which browsers will do with 301 redirects, but won't
+        // do with meta refresh redirects.
+        // https://docs.aws.amazon.com/AmazonS3/latest/dev/how-to-page-redirect.html
+        const redirect = getMetaRefreshRedirect(filePath);
+
         const contentFile = new aws.s3.BucketObject(
             relativeFilePath,
             {
                 acl: "public-read",
                 key: relativeFilePath,
                 bucket: contentBucket,
-                source: new pulumi.asset.FileAsset(filePath),
-                contentType: getMimeType(filePath) || undefined,
+
+                // Use "/dev/null" for an empty object if this is a redirect, otherwise the actual file path.
+                source: new pulumi.asset.FileAsset(redirect ? "/dev/null" : filePath),
+
+                // Add `contentType` if this isn't a redirect.
+                ...!redirect && { contentType: getMimeType(filePath) },
+
+                // Add `websiteRedirect` if this is a redirect.
+                ...redirect && { websiteRedirect: redirect },
             },
             {
                 parent: contentBucket,
             });
     });
 
+// Returns the redirect URL if filePath is a .html file that contains a meta refresh redirect, otherwise undefined.
+function getMetaRefreshRedirect(filePath: string): string | undefined {
+    // Only .html files contain meta refresh redirects.
+    if (path.extname(filePath) !== ".html") {
+        return undefined;
+    }
+
+    // Extract the redirect from the content of the file.
+    const text = fs.readFileSync(filePath, "utf8");
+    const regex = /<meta\s+?http-equiv="refresh"\s+?content="0;\s+?url=(.*?)"/gm;
```

Comment:
It might be worth-while to change this to be a case-_insensitive_ search, using `/gmi`. While the meta redirects are generated by Hugo and like "safe", it would just make this method a bit more robust.

Reply:
Done.